datalad/core/local/configuration.py

```diff
@@ -0,0 +1,337 @@
+                if have_subds else '',
+            )
+        else:
+            prefix = ''
+
+        if kwargs.get('action', None) == 'dump':
+            ui.message('{}{}={}'.format(
+                prefix,
+                ac.color_word(name, ac.BOLD),
+                res['value'] if res['value'] is not None else '',
+            ))
+        else:
+            ui.message('{}{}'.format(
+                prefix,
+                res['value'] if res['value'] is not None else '',
+            ))
+
+
+def configuration(action, scope, specs, res_kwargs, ds=None):
+    if scope == 'global':
+        cfg = dlcfg
+    else:
+        cfg = ds.config
+
+    if action not in config_actions:
+        raise ValueError("Unsupported action '{}'".format(action))
+
+    if action == 'dump':
+        if specs:
+            lgr.warning('Provided configuration selection is ignored when dumping')
+        # dumping is querying for all known keys
+        # TODO warn that specs will be ignored, if there are any
+        specs = [(n,) for n in sorted(set(cfg_defs.keys()).union(cfg.keys()))]
+        scope = None
+
+    for spec in specs:
+        if '.' not in spec[0]:
+            yield get_status_dict(
+                res_kwargs,
+                ds=ds,
+                logger=lgr,
+                status='error',
+                message=(
+                    "Configuration key without a section: '%s'",
+                    spec[0],
+                )
+            )
+            continue
+        # TODO without get-all there is little sense in having add
+        #if action == 'add':
+        #    res = _add(cfg, scope, spec)
+        if action == 'get':
+            res = _get(cfg, scope, spec[0])
+        elif action == 'dump':
+            res = _dump(cfg, spec[0])
+        # TODO this should be there, if we want to be comprehensive
+        # however, we turned this off by default in the config manager
+        # because we hardly use it, and the handling in ConfigManager
+        # is not really well done.
+        #elif action == 'get-all':
+        #    res = _get_all(cfg, scope, spec)
+        elif action == 'set':
+            res = _set(cfg, scope, *spec)
+        elif action == 'unset':
+            res = _unset(cfg, scope, spec[0])
+
+        if ds:
+            res['path'] = ds.path
+
+        if 'status' not in res:
+            res['status'] = 'ok'
+
+        yield dict(res_kwargs, **res)
+
+    if action in ('add', 'set', 'unset'):
+        # we perform a single reload, rather than one for each modification
+        # TODO: can we detect a call from cmdline? We could skip the reload.
+        cfg.reload(force=True)
+
+
+def _dump(cfg, name):
+    value = cfg.get(
+        name,
+        # pull a default from the config definitions
+        # if we have no value, but a key (i.e. in dump mode)
+        cfg_defs.get(name, {}).get('default', None))
+    return dict(
+        action='get_configuration',
```

Comment:
Should the action be 'dump_configuration' for consistency?

Reply:
Yes, done.